pkg/cmd/repo/autolink/view/http_test.go

```diff
@@ -0,0 +1,102 @@
+package view
+
+import (
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/cli/cli/v2/internal/ghrepo"
+	"github.com/cli/cli/v2/pkg/cmd/repo/autolink/shared"
+	"github.com/cli/cli/v2/pkg/httpmock"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestAutolinkViewer_View(t *testing.T) {
+	repo := ghrepo.New("OWNER", "REPO")
+
+	tests := []struct {
+		name         string
+		id           string
+		stubStatus   int
+		stubRespJSON string
+
+		expectedAutolink *shared.Autolink
+		expectErr        bool
+		expectedErrMsg   string
+	}{
+		{
+			name:       "200 successful alphanumeric view",
+			id:         "123",
+			stubStatus: 200,
+			stubRespJSON: `{
+				"id": 123,
+				"key_prefix": "TICKET-",
+				"url_template": "https://example.com/TICKET?query=<num>",
+				"is_alphanumeric": true
+			}`,
+			expectedAutolink: &shared.Autolink{
+				ID:             123,
+				KeyPrefix:      "TICKET-",
+				URLTemplate:    "https://example.com/TICKET?query=<num>",
+				IsAlphanumeric: true,
+			},
+		},
```

Comment:
The tests here are on point!  Wanting to include the numeric scenario and DRY it up a little, I pushed a couple commits as the resulting suggestion was a little large and difficult to work with in the UI. 🙇